wp_api/src/wp_com/support_tickets.rs

```diff
@@ -0,0 +1,145 @@
+use std::collections::HashMap;
+
+use serde::{Deserialize, Serialize};
+
+use crate::{date::WpGmtDateTime, impl_as_query_value_for_new_type};
+
+#[derive(Debug, PartialEq, Eq, Serialize, uniffi::Record)]
+pub struct CreateSupportTicketParams {
+    pub subject: String,
+    pub message: String,
+    pub application: String,
+    #[uniffi(default = None)]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub wpcom_site_id: Option<u64>,
+    #[uniffi(default = [])]
+    pub tags: Vec<String>,
+    #[uniffi(default = [])]
+    pub attachments: Vec<String>,
+}
+
+#[derive(Debug, PartialEq, Eq, Serialize, Deserialize, uniffi::Record)]
+pub struct SupportConversationSummary {
+    pub id: ConversationId,
+    pub title: String,
+    pub description: String,
+    pub status: String,
+    pub created_at: WpGmtDateTime,
+    pub updated_at: WpGmtDateTime,
+}
+
+#[derive(Debug, PartialEq, Eq, Serialize, Deserialize, uniffi::Record)]
+pub struct SupportConversation {
+    pub id: ConversationId,
+    pub title: String,
+    pub description: String,
+    pub status: String,
+    pub created_at: WpGmtDateTime,
+    pub updated_at: WpGmtDateTime,
+    pub messages: Vec<SupportMessage>,
+}
+
+#[derive(Debug, PartialEq, Eq, Serialize, Deserialize, uniffi::Record)]
+pub struct SupportMessage {
+    pub id: u64,
+    pub content: String,
+    pub author: SupportMessageAuthor,
+    pub role: String,
+    pub author_is_current_user: bool,
+    pub created_at: WpGmtDateTime,
+    pub attachments: Vec<SupportAttachment>,
+}
+
+#[derive(Debug, PartialEq, Eq, Serialize, Deserialize, uniffi::Record)]
+pub struct SupportUserIdentity {
+    pub id: u64,
+    pub email: String,
+    pub display_name: String,
+    pub avatar_url: String,
+}
+
+#[derive(Debug, PartialEq, Eq, Serialize, Deserialize, uniffi::Record)]
+pub struct SupportAttachment {
+    pub id: u64,
+    pub filename: String,
+    pub content_type: String,
+    pub size: u64,
+    pub url: String,
+    pub metadata: HashMap<String, AttachmentMetadataValue>,
+}
+
+#[derive(Debug, PartialEq, Eq, Serialize, Deserialize, uniffi::Enum)]
+#[serde(untagged)]
+pub enum SupportMessageAuthor {
+    User(SupportUserIdentity),
+    SupportAgent(SupportAgentIdentity),
```

Comment:
Should there be an "other" variant, just in case the "role" value is some other value, like "system" messages?

Reply:
My concern is that the app may fail to parse the messages if the API introduces a new "author" with a different JSON schema than "user" and "agent".

Reply:
Let me think about this again depending on your answer to [this](https://github.com/Automattic/wordpress-rs/pull/743#discussion_r2136596892).

My general worry is for a parsing issue to go unnoticed and introducing a fallback inherently has that problem. Ideally I'd like to have a way to not have parsing errors, but still get a notice about the unexpected fallbacks. Once we have logging implemented, it might help with that a bit.